psydac/linalg/block.py

```diff
@@ -292,6 +293,22 @@ def __getitem__( self, key ):
     def __setitem__( self, key, value ):
         assert value.space == self.space[key]
         self._blocks[key] = value
+    def conjugate( self, out=None):
+        if out is not None:
+            assert isinstance( out, BlockVector )
+            assert out.space is self.space
+
+        else:
+            out = self.copy()
+
+        for Lij in out.blocks:
+            assert isinstance( Lij, StencilVector )
+            Lij.conjugate(out=Lij)
+
+        return out
```

Comment:
I think that all blocks should be `StencilVector` because it is the only one to have a `conjugate()` method.

Maybe it can also be a `BlockVector`

Reply:
If you think that the `conjugate` method is useful in general, we should require every `Vector` subclass to provide it by adding an `@abstractmethod` to the base class.

CC: @jowezarek @e-moral-sanchez @saidctb @campospinto

Reply:
[5449ed2](https://github.com/pyccel/psydac/pull/273/commits/5449ed295a32548ef01dd6bfa0928688fd87de53)

Reply:
I am not sure of what changes you have made here. I was suggesting that the `Vector` base class (in `psydac.linalg.basic`) should have this method:
```python
@abstractmethod
def conjugate(self, out=None):
    """Compute the complex conjugate vector.

    If the field is real (i.e. `self.dtype in (np.float32, np.float64)`) this method is equivalent to `copy`.
    If the field is complex (i.e. `self.dtype in (np.complex64, np.complex128)`) this method returns
    the complex conjugate of `self`, element-wise.

    The behavior of this function is similar to `numpy.conjugate(self, out=None)`.
    """
```

Reply:
@tomcaruso 👋

Reply:
This pull request add the abstract property conjugate in Vector
[c24f4c4](https://github.com/pyccel/psydac/pull/273/commits/c24f4c49f0f175ca823e3bf9da53d35d13f6dde0)
And this pull request add the conjugate property for the DenseVector
[491aba8](https://github.com/pyccel/psydac/pull/273/commits/491aba8d9492272cc994d428a6412e4b520222fb)